src/development/building-and-debugging.md

```diff
@@ -0,0 +1,38 @@
+
+# Building and Debugging the library crates
+
+Most of the [instructions from the rustc-dev-guide][rustc-guide] also apply to the standard library since
+it is built with the same build system, so it is recommended to read it first.
+
+[rustc-guide]: https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html
+
+# Println-debugging alloc and core
+
+Since logging and IO APIs are not available in `alloc` and `core` advice meant for the rest of the compiler
+is not applicable here.
+
+Instead one can either extract the code that should be tested to a normal crate and add debugging statements there or
+on POSIX systems one can use the following hack:
+
+```rust
+extern "C" {
+    fn dprintf(fd: i32, s: *const u8, ...);
+}
+
+macro_rules! dbg_printf {
+    ($s:expr) => {
+        unsafe { dprintf(2, $s as *const u8); }
```

Comment:
This should pass `"%s\0".as_ptr()` as the first argument, and `$s` as the second. Otherwise you'll have issues if you try to log something with percent signs.